core/script_debugger_remote.cpp

```diff
@@ -598,7 +598,12 @@ void ScriptDebuggerRemote::_send_object_id(ObjectID p_id) {
 	Array send_props;
 	for (int i = 0; i < properties.size(); i++) {
 		const PropertyInfo &pi = properties[i].first;
-		const Variant &var = properties[i].second;
+		Variant &var = properties[i].second;
+
+		if (WeakRef * ref = Object::cast_to<WeakRef>(var)) {
```

Comment:
We try to avoid using assignments as conditions, so it would be better to cast before the if and then check `if (ref)`.

Reply:
Why not using assignments?
The `ref` shouldn't be available out of this condition block.

Reply:
It's mostly a matter of style, and to ensure that we don't end up with bogus `if (existing_var = test)` conditions when an equality test is meant (`==`). If we don't allow assignments in `if`, we know that any single `=` should raise a warning when we read the code.

Reply:
Fine.